packages/x-components/src/wiring/__tests__/wires-factory.spec.ts

```diff
@@ -186,4 +189,129 @@ describe('testing wires factory', () => {
+      public oneParameterMethod = jest.fn((parameter: string) => parameter);
+      public restParametersMethod = jest.fn((...parameters: number[]) => parameters);
+      public multipleParametersMethod = jest.fn((a: number, b: number) => a + b);
+    }
+
+    describe(`testing ${wireService.name}`, () => {
+      it('allows calling methods with no parameters', () => {
+        const service = new TestService();
+        const wireTestService = wireService(service);
+        const wire = wireTestService('noParametersMethod');
+        wire(subjectHandler.subject, storeMock, busOnMock);
+        subjectHandler.emit('something');
+
+        expect(service.noParametersMethod).toHaveBeenCalledTimes(1);
+        // Methods that have no parameters will be invoked with the payload.
+        expect(service.noParametersMethod).toHaveBeenCalledWith('something');
+      });
+
+      it('allows calling methods with one parameter', () => {
+        const service = new TestService();
+        const wireTestService = wireService(service);
+        const wire = wireTestService('oneParameterMethod');
+        wire(subjectHandler.subject, storeMock, busOnMock);
+        subjectHandler.emit('milk');
+
+        expect(service.oneParameterMethod).toHaveBeenCalledTimes(1);
+        expect(service.oneParameterMethod).toHaveBeenCalledWith('milk');
+      });
+
+      it('allows calling methods with one parameter passed statically', () => {
+        const service = new TestService();
+        const wireTestService = wireService(service);
+        const wire = wireTestService('oneParameterMethod', 'torreznos');
+        wire(subjectHandler.subject, storeMock, busOnMock);
+        subjectHandler.emit('lettuce');
+
+        expect(service.oneParameterMethod).toHaveBeenCalledTimes(1);
+        expect(service.oneParameterMethod).toHaveBeenCalledWith('torreznos');
+      });
+
+      it('allows calling methods with rest parameters', () => {
+        const service = new TestService();
+        const wireTestService = wireService(service);
+        const wire = wireTestService('restParametersMethod');
+        wire(subjectHandler.subject, storeMock, busOnMock);
+        subjectHandler.emit(10);
+
+        expect(service.restParametersMethod).toHaveBeenCalledTimes(1);
+        expect(service.restParametersMethod).toHaveBeenCalledWith(10);
+      });
+
+      it('allows calling methods with rest parameters passed statically', () => {
+        const service = new TestService();
+        const wireTestService = wireService(service);
+        const wire = wireTestService('restParametersMethod', 5);
+        wire(subjectHandler.subject, storeMock, busOnMock);
+        subjectHandler.emit(10);
+
+        expect(service.restParametersMethod).toHaveBeenCalledTimes(1);
+        expect(service.restParametersMethod).toHaveBeenCalledWith(5);
+      });
+
+      // eslint-disable-next-line jest/expect-expect
+      it("doesn't allow calling methods with multiple parameters", () => {
```

Comment:
I got the idea of this test and the other one below  but I'm not sure if we should use it.

I mean this is something more related with Typescript and the good use of their types.

Reply:
This is something that was not possible before without too much code, or complex tricks. I know this test doesn't have any jest assertion, but I think it adds confidence to TS code.